File: src/2778.sum-of-squares-of-special-elements.2844/solution.ts
function sumOfSquares (nums: number[]): number {
    let result = 0;
    const L = nums.length;
    for (let i = 1; i <= Math.sqrt(L); i++) {
        if (L % i !== 0) {
            continue;
        }
        const rest = L / i;
        result += Math.pow(nums[i - 1], 2);

        if (rest !== i) {
            result += Math.pow(nums[rest - 1], 2);
        }
    }
    return result;
}

File: src/2778.sum-of-squares-of-special-elements.2844/solution.java
class Solution {
    public int sumOfSquares(int[] nums) {
        int result = 0;
        int N = nums.length;
        for (int i = 0; i < nums.length; i++) {
            if (N % (i + 1) == 0) {
                result += nums[i] * nums[i];
            }
        }
        return result;
    }
}